services/core-web/cypress/e2e/majorprojects.cy.ts

```diff
@@ -0,0 +1,86 @@
+describe("Major Projects", () => {
+    beforeEach(() => {
+        cy.login();
+        cy.wait(5000);
+
+        // Get the button element by its text content
+        cy.contains('button', 'Major Projects')
+            // Check that it is visible and enabled
+            .should('be.visible')
+            .and('not.be.disabled')
+            // Click on it
+            .click({ force: true });
+
+        // Find the table and wait for it to be visible
+        cy.get('table[style="table-layout: auto;"]').should('be.visible').find('tbody > tr').eq(1) // get the second row
```

Comment:
Is there another selector we can use besides `table[style="table-layout: auto;"]`? I worry that selectors like that are quite brittle. I am okay with adding in IDs, data attributes for cypress to find things, but styles and classes defined by libraries are dicey.

Reply:
Totally agree.  I think adding ids for the sake of testing will make them both more durable and more readable.

Reply:
Not to mention making it easier to write the tests

Reply:
I have incorporated the data-cy attribute into the table to streamline its testing process with Cypress.